PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -274,25 +270,28 @@ + (NSMutableDictionary *)messageFromData:(id)data
 
 + (NSMutableDictionary *)presenceFromData:(NSDictionary *)data {
     
+#warning ask about how this is affected
+    NSDictionary *payload = data[@"d"];
+    
     NSMutableDictionary *presence = [NSMutableDictionary new];
     
     // Processing common for all presence events data.
-    presence[@"presenceEvent"] = (data[@"action"]?: @"interval");
+    presence[@"presenceEvent"] = (payload[@"action"]?: @"interval");
     presence[@"presence"] = [NSMutableDictionary new];
-    presence[@"presence"][@"timetoken"] = data[@"timestamp"];
-    if (data[@"uuid"]) {
+    presence[@"presence"][@"timetoken"] = payload[@"timestamp"];
+    if (payload[@"uuid"]) {
         
-        presence[@"presence"][@"uuid"] = data[@"uuid"];
+        presence[@"presence"][@"uuid"] = payload[@"uuid"];
     }
     
     // Check whether this is not state modification event.
     if (![presence[@"presenceEvent"] isEqualToString:@"state-change"]) {
         
-        presence[@"presence"][@"occupancy"] = (data[@"occupancy"]?: @0);
+        presence[@"presence"][@"occupancy"] = (payload[@"occupancy"]?: @0);
     }
-    if (data[@"data"]) {
+    if (payload[@"data"]) {
      
-        presence[@"presence"][@"state"] = data[@"data"];
+        presence[@"presence"][@"state"] = payload[@"data"];
```

Comment:
@jzucker2 Because data[kPNMessagePayloadDataKey] will be passed from calling method, there is no need to extract presence event data into separate variable and original code can be used with _data_ instead of _payload_.